core/src/main/java/io/github/tcdl/msb/message/MessageFactory.java

```diff
@@ -89,4 +98,21 @@ private String createCorrelationId(Message originalMessage) {
             return Utils.generateId();
         }
     }
+
+    private Set<String> createTags(MessageTemplate messageTemplate, Message originalMessage) {
+        List<String> tags = new ArrayList<>();
```

Comment:
java 8 way to do this is http://winterbe.com/posts/2015/03/15/avoid-null-checks-in-java/